framework_tvb/tvb/core/entities/file/files_update_manager.py

```diff
@@ -195,17 +226,34 @@ def run_all_updates(self):
             no_ignored = 0
             start_time = datetime.now()
 
-            # Read DataTypes in pages to limit the memory consumption
-            for current_idx in range(0, total_count, self.DATA_TYPES_PAGE_SIZE):
-                datatypes_for_page = dao.get_all_datatypes(current_idx, self.DATA_TYPES_PAGE_SIZE)
-                count_ok, count_error, count_ignored = self.__upgrade_datatype_list(datatypes_for_page)
-                no_ok += count_ok
-                no_error += count_error
-                no_ignored += count_ignored
+            if TvbProfile.current.version.DATA_CHECKED_TO_VERSION < 4:
+                # Read DataTypes in pages to limit the memory consumption
+                for current_idx in range(0, total_count, self.DATA_TYPES_PAGE_SIZE):
+                    datatypes_for_page = dao.get_all_datatypes(current_idx, self.DATA_TYPES_PAGE_SIZE)
+                    count_ok, count_error, count_ignored = self.__upgrade_datatype_list(datatypes_for_page)
+                    no_ok += count_ok
+                    no_error += count_error
+                    no_ignored += count_ignored
+
+                    self.log.info(
+                        "Updated H5 files so far: %d [fine:%d, error:%d, ignored:%d of total:%d, in: %s min]" % (
+                            current_idx + len(datatypes_for_page), no_ok, no_error, no_ignored, total_count,
+                            int((datetime.now() - start_time).seconds / 60)))
+            else:
+                # Migrating the configuration file
+                new_stored_settings = {KEY_ADMIN_DISPLAY_NAME: 'Administrator', KEY_ENABLE_KC_LOGIN: False,
+                                       KEY_KC_WEB_CONFIGURATION: '', KEY_KC_CONFIGURATION: 'add_keycloak_path_here'}
+                manager = TvbProfile.current.manager
+                manager.add_entries_to_config_file(new_stored_settings)
```

Comment:
Instead of the need to add these new Settings, can't we have some defaults for these in the service/config where these are Read first?

Reply:
I added a new upgrade_code script and moved the code there.